core/src/main/java/org/pentaho/di/core/plugins/PluginRegistry.java

```diff
@@ -1063,4 +1064,26 @@ public void reset() {
       lock.writeLock().unlock();
     }
   }
+
+  public PluginInterface findPluginWithId( Class<? extends PluginTypeInterface> pluginType, String pluginId, boolean waitForPluginToBeAvailable ) {
+    return waitForPluginToBeAvailable && findPluginWithId(  pluginType,  pluginId ) == null
+      ? waitForPluginToBeAvailable( pluginType,  pluginId, waitForPluginToBeAvailableLimit )
+      : findPluginWithId(  pluginType,  pluginId );
+  }
+
+  public PluginInterface waitForPluginToBeAvailable( Class<? extends PluginTypeInterface> pluginType, String pluginId, int waitLimit ) {
+    int timeToSleep = 1;
+    try {
+      Thread.sleep( timeToSleep );
+      waitLimit -= timeToSleep;
+    } catch ( InterruptedException e ) {
+      throw new RuntimeException( "Waiting for Plugin to be Available interrupted!", e );
```

Comment:
Need to reset the interrupt flag (with Thread.currentThread().interrupt())
Also, I'd suggest logging the interruptedexception and returning null rather than rethrowing.  Or, if you need to rethrow, use a less generic runtime exception - maybe IllegalStateException.